src/java/org/apache/cassandra/db/commitlog/CommitLogSegmentManagerCDC.java

```diff
@@ -78,11 +81,45 @@ public void discard(CommitLogSegment segment, boolean delete)
                 FileUtils.deleteWithConfirm(cdcLink);
 
             File cdcIndexFile = segment.getCDCIndexFile();
-            if (cdcIndexFile.exists())
-                FileUtils.deleteWithConfirm(cdcIndexFile);
+            deleteCDCFiles(cdcLink, cdcIndexFile);
         }
     }
 
+    /**
+     * Delete the oldest CDC commit log segment to free up space.
+     * @return total deleted file size in bytes
+     */
+    public long deleteOldestCommitLogSegment()
```

Comment:
Strongly recommend renaming this to `deleteOldestLinkedCDCSegmentFile` or something. It took me a double take of being Very Scared of this method before realizing it was deleting the oldest CDC segment and not just wildly deleting old CommitLogSegment files from the system. 😀

Reply:
👍 Sure. Let's rename the method to add clarity. I probably omitted "CDC" for it is in the  `CommitLogSegmentManagerCDC` class.